features/enlinkd/service/api/src/main/java/org/opennms/netmgt/enlinkd/service/api/SubNetwork.java

```diff
@@ -0,0 +1,146 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2013-2014 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2014 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.enlinkd.service.api;
+
+import java.net.InetAddress;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.opennms.core.utils.InetAddressUtils;
+import org.opennms.netmgt.enlinkd.model.IpInterfaceTopologyEntity;
+import org.springframework.util.Assert;
+
+public class SubNetwork {
```

Comment:
This whole class doesn't look thread-safe at all, and it seems possible that these things could be accessed concurrently depending on how scanning goes. Perhaps some read/write locking should be implemented around the node/interface map access?

If they'll only be accessed internally to one thread, then feel free to mark this as resolved. :)

Reply:
Ben, 
 actually this class is only used by EnhancedLinkd. 
 How it works...instead of scheduling collection at startup I have created Scheduled a group for every Protocol.
 So when the Thread executes on scheduled group it is created a list of "Executable" that will perform data collection, each with his own priority and send to the LegacyExecutorService.
 
 the priority is inherited from the subnetworks. Each node on the same network will have the same priority.
 
 The priorityMap is build calling the NideTopologyService.
 
 I priority Map is created for each thread and is accessed only by the thread.